queries/utility/numbers_grammar.py

```diff
@@ -0,0 +1,184 @@
+"""
+
+    Greynir: Natural language processing for Icelandic
+
+    Number parsing grammar.
+
+    Copyright (C) 2022 Miðeind ehf.
+
+       This program is free software: you can redistribute it and/or modify
+       it under the terms of the GNU General Public License as published by
+       the Free Software Foundation, either version 3 of the License, or
+       (at your option) any later version.
+       This program is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see http://www.gnu.org/licenses/.
+
+
+    Utility module
+    Exposes nonterminal "UHeilTala" for parsing
+    numbers either written in natural language or in digits,
+    along with "UBrotaTala" for parsing floats ("(number) 'komma' (numbers)")
+    written in natural language.
+    Returns the number values in the list result["numbers"].
+
+"""
+
+# TODO: Deal better with cases and genders of numbers
+# TODO: Allow "tólf hundruð þúsund" & "hundruðir" (need to add hundruðir to ord.add/auka.csv)
```

Comment:
"hundruðir" is technically wrong (see http://malfar.arnastofnun.is/grein/65992) but I guess we should allow it? It's a very common error.

Reply:
Yeah, I was thinking that too.

Reply:
Good thing you got me thinking about this. Turns out Yfirlestur doesn't catch "hundruðir"!

<img width="1096" alt="Screen Shot 2022-10-27 at 14 13 39" src="https://user-images.githubusercontent.com/654045/198309803-0d0ff615-c5dc-4714-afbd-3c9db6af80e8.png">